### ISSUE TITLE
Question about MessageStream.Inbound's consumer

### ISSUE THREAD

**Issue by coaku** [jonstout/ogo#31]
hi @jonstout,

As far as I can see, after a switch starts up, function `Controller.handleConnection()` and `OFSwitch.receive()` are both tring to pop messages from the same inbound channel `OFSwitch.stream.Inbound`. However the former one can only deal with the OPFT_HELLO or OFPT_FEATURES_REPLY messages.

So, other messages will be discarded if received by `Controller.handleConnection()` ?

**Comment by jonstout** [comment]
Correct. `Controller.handleConnection` only concerns itself with the openflow channel setup. Until the openflow handshake is complete everything else is ignored. Once the dpid is discovered using the `ofp_features_request` message the connection stream is passed along to a Switch instance.
